src/Plugin/CookiePlugin.php

```diff
@@ -69,7 +69,7 @@ public function handleRequest(RequestInterface $request, callable $next, callabl
                     }
 
                     // Restrict setting cookie from another domain
-                    if (false === strpos($cookie->getDomain(), $request->getUri()->getHost())) {
+                    if (false === strpos($request->getUri()->getHost(), $cookie->getDomain())) {
```

Comment:
@dbu, @sagikazarmark:
What if we add a dot `.` to both cookie domain and request domain before the comparison?
```php
strpos(
    '.'.$request->getUri()->getHost(),
    '.'.$cookie->getDomain()
)
```
Here's what's happening in this case:
```php
strpos(
    '.hacksymfony.slack.com',
    '.symfony.slack.com'
); // no match, cookie ignored.

strpos(
    '.hack.symfony.slack.com',
    '.symfony.slack.com'
); // match, cookie used.
```
I wonder if this workaround is appropriate and not counted as dirty code.